domdiv/main.py

```diff
@@ -560,6 +583,57 @@ def read_card_data(options):
         Card.sets[s]['no_randomizer'] = Card.sets[s].get('no_randomizer', False)
         Card.sets[s]['fan'] = Card.sets[s].get('fan', False)
 
+    # Remove the Trash card. Do early before propagating to various sets.
+    if options.no_trash:
+        i = find_index_in_list(cards, 'card_tag', 'Trash')
+        if i is not None:
+            del cards[i]
+
+    # Repackage Curse cards into 10 per divider. Do early before propagating to various sets.
+    if options.curse10:
+        i = find_index_in_list(cards, 'card_tag', 'Curse')
+        if i is not None:
+            new_cards = []
+            cards_remaining = cards[i].getCardCount()
+            while cards_remaining > 10:
+                # make a new copy of the card and set count to 10
+                new_card = copy.deepcopy(cards[i])
+                new_card.setCardCount(10)
+                new_cards.append(new_card)
+                cards_remaining -= 10
+
+            # Adjust original Curse card to the remaining cards (should be 10)
+            cards[i].setCardCount(cards_remaining)
+            # Add the new dividers
+            cards.extend(new_cards)
+
+    # Create Start Deck dividers. 4 sets. Adjust totals for other cards, too.
+    # The card database contains one prototype divider that needs to be either duplicated or deleted.
+    # Do early before propagating to various sets.
+    i = find_index_in_list(cards, 'card_tag', 'Start Deck')
+    if i is not None:
+        new_cards = []
+        if options.start_decks:
+            # Get the index for Copper and Estate cards so their totals can be adjusted
+            c_i = find_index_in_list(cards, 'card_tag', 'Copper')
+            e_i = find_index_in_list(cards, 'card_tag', 'Estate')
+            if c_i is not None and e_i is not None:
+                # Take card counts out of Copper and Estate for 4 Start Decks
+                cards[c_i].setCardCount(cards[c_i].getCardCount() - (4 * 7))
+                cards[e_i].setCardCount(cards[e_i].getCardCount() - (4 * 3))
+                # Add cards to Start Deck prototype
+                cards[i].setCardCount(7)
+                cards[i].addCardCount([int(3)])
+                for x in range(0, 3):
```

Comment:
See above, should depend on number of starting decks specified.

Reply:
I actually started to make the number of starting decks an option, but then stopped after I thought about it a bit. 

1 to 4 is easily done.  But remember that this applies to all the base cards.  So let's say the user picks 2.  So it makes 2 start decks for Dominion 1st edition,  2 start decks for Dominion 2nd edition, 2 decks for Intrigue 1st edition, (none for 2nd addition), and 2 start decks for the Base.  So you really don't get 2, you get 2 x the number of expansions that you include that contain base cards.  So two if not really 2.

The harder case is picking something greater than 4.  After 4, a given set does not have enough Copper and Estate cards.  So where do those come from?  From a slip case point of view, we need to keep the counts so the slip cases are the right depth.  And we never really know which expansions the user will include.   I know it could be done, but even more options/information would be needed from the user to do it "right" for them (which might be different from others.)
The same problem occurs if you really want "2" to mean "2" above.  From which expansion do we make these starting decks from?

In the end, I settled on the fact that the game is designed for 1 to 4 players each with starting decks.  The number of Estates used in a 3 and 4 player game (12) matches exactly the number that is left over after you make 4 starting decks.  No special counting.   While the total number of starting decks is 4x the number of expansions with base cards, in the end,  I figured if they want starting decks they will most likely be consistent and want them across the board.  And after a while, these cards are extras anyway and it keeps them from having to remove the extra Estates.

Reply:
Yeah, ok, you're probably right. Let's leave it as you have it.

Reply:
So I thought about this some more.  And maybe there is a middle ground.

If we limit the starting decks to only the Base (and not included in Dominion, et. al. with the `--base-cards-with-expansion`) and then make the option `--start-decks` accept 0 - 4, I think this may work. 

If someone wants 6 starting decks, they have to run the program twice, once with 4 and once with 2.  I guess they could do 3 and 3, but the Estate counts will be weird (but correct!).  In any case, for that printed Base set, the card counts would be correct.

The only oddity is if they are trying to make base starting decks and they don't have the cards from the other sets.  They would just have to substitute the Start Decks, Copper, and Estate dividers from the Base into their Dominion set - it would not be printed out exactly as used with the `--base-cards-with-expansion`.

Reply:
For now, I left the above option out and have keep this to doing 4 start decks.  But I did put in some better infrastructure if we do want to change to a variable number of start decks.  The biggest change was allowing for any number of attributes to be matched in `find_index_of_object`

If we do change:
* the location this code would need to change to AFTER the cards are expanded out for all the sets
* The card index searches with `find_index_of_object` would need to add a 'cardset': 'base' attribute
* and the argument would need to take an integer (default to 0) which would be used in the copy loop.
Then everything is localised to only the base set.  Not to any base cards added to other sets.

Reply:
For some reason, I'm still getting CI failures.   It passes on my local system.  Has anything changed?

All are "W605 invalid escape sequence" in draw.py and cards.py